crates/orchestrator/src/tests/database/mod.rs

```diff
@@ -158,50 +161,42 @@ async fn database_get_jobs_after_internal_id_by_job_type_works() {
     assert_eq!(jobs_after_internal_id[1], job_vec[5]);
 }
 
-/// Test for `update_job_status` operation in database trait.
-/// Happy Case : Creating a job with version 0 and updating the job with version 0 update only.
 #[rstest]
 #[tokio::test]
-async fn database_update_job_status_passing_case_works() {
+async fn database_test_update_job() {
     let services = TestConfigBuilder::new().configure_database(ConfigType::Actual).build().await;
     let config = services.config;
-
     let database_client = config.database();
 
-    let job = build_job_item(JobType::SnosRun, JobStatus::Created, 1);
-
+    let job = build_job_item(JobType::DataSubmission, JobStatus::Created, 456);
     database_client.create_job(job.clone()).await.unwrap();
 
-    let updating_job_res = database_client.update_job_status(&job, JobStatus::Completed).await.is_ok();
-
-    assert!(updating_job_res, "Job result assertion failed");
-}
-
-/// Test for `update_job_status` operation in database trait.
-/// Failing Case : Creating a job with version 1 and updating the job with version 0 update only.
-#[rstest]
-#[tokio::test]
-async fn database_update_job_status_failing_case_works() {
-    let services = TestConfigBuilder::new().configure_database(ConfigType::Actual).build().await;
-    let config = services.config;
-    let database_client = config.database();
-
-    // Scenario :
-
-    // Worker 1 :
-    // Job is created
-    let job = build_job_item(JobType::SnosRun, JobStatus::Created, 1);
-    database_client.create_job(job.clone()).await.unwrap();
-
-    // Worker 2 :
-    // Job is updated
-    let updating_job_res = database_client.update_job_status(&job, JobStatus::Completed).await.is_ok();
-    assert!(updating_job_res, "Job result assertion failed");
-
-    // Worker 1 :
-    // Job update try (update should fail)
-    let updating_job_res = database_client.update_job_status(&job, JobStatus::PendingVerification).await.is_err();
-    assert!(updating_job_res, "Job result assertion failed");
+    let job_id = job.id;
+
+    let metadata = HashMap::new();
+    let key = "test_key";
+    let updated_metadata = increment_key_in_metadata(&metadata, key).unwrap();
+
+    let job_cloned = job.clone();
+    let _ = database_client
+        .update_job(&job_cloned, JobItemUpdates {
+            status: Some(JobStatus::LockedForProcessing),
+            metadata: Some(updated_metadata),
+            external_id: None,
+            internal_id: None,
+            job_type: None,
+        })
+        .await;
+
+    let mut job_after_updates = build_job_item(JobType::DataSubmission, JobStatus::LockedForProcessing, 456);
```

Comment:
nitpick: it's cleaner if we edit the original job here instead of building a new one? or maybe we hardcode `asserts` to the new values and for old values we take them from `job`

Reply:
Alright, cleaned [here](https://github.com/madara-alliance/madara-orchestrator/pull/134/commits/a9943aeacfce70762692a00182f7f0562a44ff24).